src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,159 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.UserRegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final StorageDaoImpl storageDao = new StorageDaoImpl();
+    private final RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+    private String expectedMessage;
+    private User newUser;
+
+    @BeforeEach
+    void init() {
+        newUser = new User();
+        Storage.people.clear();
+    }
+
+    @AfterEach
+    void teardown() {
+        newUser = null;
+    }
+
+    @Test
+    @DisplayName("Method register(User user) "
+            + "throws UserRegistrationException if user parameter is null")
+    void register_nullUser_notOk() {
+        expectedMessage = "User can`t be null";
+        UserRegistrationException expectedException = assertThrows(
+                UserRegistrationException.class,
+                () -> registrationService.register(null)
+        );
+        assertEquals(
+                expectedMessage,
+                expectedException.getMessage()
+        );
+    }
+
+    @Test
+    @DisplayName("Method register(User user) "
+            + "throws UserRegistrationException if user login is null")
+    void register_nullUserLogin_notOk() {
+        expectedMessage = "User login can`t be null";
+        UserRegistrationException expectedException = assertThrows(
+                UserRegistrationException.class,
+                () -> registrationService.register(newUser)
+        );
+        assertEquals(
+                expectedMessage,
+                expectedException.getMessage()
+        );
+    }
+
+    @Test
+    @DisplayName("Method register(User user) "
+            + "throws UserRegistrationException if user password is null")
+    void register_nullUserPassword_notOk() {
+        newUser.setLogin("valid login");
+        expectedMessage = "User password can`t be null";
+        UserRegistrationException expectedException = assertThrows(
+                UserRegistrationException.class,
+                () -> registrationService.register(newUser)
+        );
+        assertEquals(
+                expectedMessage,
+                expectedException.getMessage()
+        );
+    }
+
+    @Test
+    @DisplayName("Method register(User user) "
+            + "throws UserRegistrationException if user login length less then 6 characters")
+    void register_UserLoginLengthLessThenSixCharacters_notOk() {
+        newUser.setLogin("test");
+        newUser.setPassword("valid password");
```

Comment:
add those field values as constants and name them like `VALID_PASSWORD`/`INVALID_LOGIN` etc.
Same for other tests